src/main/java/org/mastodon/mamut/clustering/util/ClusterUtils.java

```diff
@@ -378,4 +382,32 @@ public static double[] getUpperTriangle( final double[][] twoDimensionalArray )
 				array[ index++ ] = twoDimensionalArray[ i ][ j ];
 		return array;
 	}
+
+	/**
+	 * Returns the names of all tag sets in the model.
+	 * @param model the model to get the tag-set model from.
+	 * @return the names of all tag sets in the model.
+	 * TODO remove after mastodon-core release 1.0.0-beta-30 and replace by TagSetUtils.getTagSetNames
+	 */
+	public static List< String > getTagSetNames( final Model model )
+	{
+		List< String > tagSetNames = new ArrayList<>();
+		model.getTagSetModel().getTagSetStructure().getTagSets().forEach( tagSet -> tagSetNames.add( tagSet.getName() ) );
+		return tagSetNames;
+	}
+
+	/**
+	 * Gets the tag value of the first spot in the given branchSpot within the given tagSet.
+	 * @param model the model to which the branch belongs
+	 * @param branchSpot the branch spot
+	 * @param tagSet the tag set
+	 * @return the tag value
+	 * TODO remove after mastodon-core release 1.0.0-beta-30 and replace by TagSetUtils.getTagValue
+	 */
+	public static String getTagValue( final Model model, final BranchSpot branchSpot, final TagSetStructure.TagSet tagSet )
```

Comment:
Please consider renaming this method to `getTagLabel`.

Reply:
Changed it.